sphinx/domains/c.py

```diff
@@ -1075,9 +1075,11 @@ def _add_anno(signode: TextElement, text: str) -> None:
             if self.restrict or self.volatile:
                 signode += addnodes.desc_sig_space()
             _add_anno(signode, 'const')
-        if self.const or self.volatile or self.restrict or len(self.attrs) > 0:
-            if self.next.require_space_after_declSpecs():
-                signode += addnodes.desc_sig_space()
+        if (
+            (self.const or self.volatile or self.restrict or len(self.attrs) > 0) and
+            self.next.require_space_after_declSpecs()
+        ):
+            signode += addnodes.desc_sig_space()
```

Comment:
Should not be changed, the original better conveys the semantics and is more readable.